src/software/geom/algorithms/signed_distance.cpp

```diff
@@ -0,0 +1,61 @@
+#include "software/geom/algorithms/signed_distance.h"
+#include "software/geom/algorithms/distance.h"
+
+#include <algorithm>
+
+double signedDistance(const Circle &first, const Point &second)
+{
+    return distance(first.origin(), second) - first.radius();
+}
+double signedDistance(const Point &first, const Circle &second)
+{
+    return signedDistance(second, first);
+}
+
+double signedDistance(const Rectangle &first, const Point &second)
+{
+    Vector centre_to_point = second - first.centre();
+    Vector abs_centre_to_point = Vector(std::abs(centre_to_point.x()), std::abs(centre_to_point.y()));
+    Vector corner_to_point =  abs_centre_to_point - (first.posXPosYCorner() - first.centre());
+    Vector clamped_corner_to_point = Vector(std::max(corner_to_point.x(), 0.0), std::max(corner_to_point.y(), 0.0));
+
+    return clamped_corner_to_point.length() + std::min(std::max(corner_to_point.x(), corner_to_point.y()), 0.0);
```

Comment:
Is this implementation based on this? https://youtu.be/62-pRVZuS5c?t=347

Reply:
All of the implementations were based on this article: https://iquilezles.org/articles/distfunctions2d/